### PR TITLE
fix(github): don't match org name with underscore

### PR DESCRIPTION
**Pull request by PhilipAbed**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by secustor — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Comment by rarkins** [comment]
Needs deconflicting

**Comment by PhilipAbed** [comment]
@rarkins its the PR u opened on top :
https://github.com/renovatebot/renovate/pull/19070

weird thing is, i remmember pushing into my branch, but i never created a PR, because you have already done 19070

**Comment by rarkins** [comment]
If you want to improve on what I committed then deconflict, otherwise close. This has more changes than mine

File: lib/modules/platform/github/massage-markdown-links.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ interface UrlMatch {
 }
 
 const urlRegex =
-  /(?:https?:)?(?:\/\/)?(?:www\.)?(?<!api\.)(?:to)?github\.com\/[-_a-z0-9]+\/[-_a-z0-9.]+\/(?:discussions|issues|pull)\/[0-9]+(?:#[-_a-z0-9]+)?/i; // TODO #12872 (?<!re) after text not matching
+  /(?:https?:)?(?:\/\/)?(?:www\.)?(?<!api\.)(?:to)?github\.com\/[-\w\d]+\/[-_.\w\d]+\/(?:discussions|issues|pull)\/\d+(?:#[-_\w\d]+)?/i; // TODO #12872 (?<!re) after text not matching
```
Comment (rarkins) [inline lib/modules/platform/github/massage-markdown-links.ts:14]: Isn't `\w` shortcut for `[A-Za-z0-9_]`?

```suggestion
  /(?:https?:)?(?:\/\/)?(?:www\.)?(?<!api\.)(?:to)?github\.com\/[-\w]+\/[-.\w]+\/(?:discussions|issues|pull)\/\d+(?:#[-\w]+)?/i; // TODO #12872 (?<!re) after text not matching
```

Comment (rarkins) [reply lib/modules/platform/github/massage-markdown-links.ts:14]: I actually prefer the current readability though

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/platform/github/massage-markdown-links.ts (hunk 1 above)
Comment (PhilipAbed) [reply lib/modules/platform/github/massage-markdown-links.ts:14]: its not accurate then, 
well how about we remove the useless `(?<!api\.)` ?

**Review by rarkins — APPROVED** [review]

**Pull request merged**